### ISSUE TITLE
Hot Reload Support

### ISSUE THREAD

**Issue by quaternion7** [godot-zig/godot-zig#25]
Using the `godot-zig-example` project, I tried setting `reloadable = true` in the gdextension definition, but godot prints this error in the console:
```
ERROR: Extension marked as reloadable, but attempted to register class 'ExampleNode' which doesn't support reloading. Perhaps your language binding don't support it? Reloading disabled for this extension.
   at: _register_extension_class_internal (core/extension/gdextension.cpp:426)
``` 
Is it a limitation of the current binding or is there something missing in the ExampleNode class?

**Comment by elvisxzhou** [comment]
Yes, currently hot reload is not supported due to the limitation of Zig shared library generation, in which I dont find a way to disable STB_GNU_UNIQUE. (-fno-gnu-unique in GCC/LLVM)

**Comment by ndbn** [comment]
At now reloading work, 4.3rc2 / 70e156b 2 months ago/ Zig 0.14.0-dev.620+eab934814
